.github/workflows/gh-pages.yml

```diff
@@ -0,0 +1,65 @@
+name: Build
+on:
+  push:
+    branches:
+      - main
+
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout project
+        uses: actions/checkout@v3
+      - name: Checkout submodules
+        run: |
+          git submodule init
+          git submodule update
+      - name: Install build dependencies
+        run: |
+          sudo apt update
+          sudo apt install hugo
+      - name: Configure hugo
+        run: |
+          patch -o config.toml config.toml.example << EOF
+          1c1
+          < baseURL = "http://localhost/"
+          ---
+          > baseURL = "https://hugonikanor.github.io/r-selfhosted-wiki"
+          EOF
+      - name: Build page
+        run: hugo
+      - name: Upload atrifacts
+        uses: actions/upload-artifact@v3
+        with:
+          path: public
+
+  prepare_page:
+    runs-on: ubuntu-latest
+    needs:
+      - build
+    steps:
+      - name: Retrieve build
+        uses: actions/download-artifact@v3
+      - name: Upload Pages Artifact
+        uses: actions/upload-pages-artifact@v1.0.8
+        with:
+          path: artifact
+
+  deploy:
+    environment:
+      name: github-pages
+      url: ${{ steps.deployment.outputs.page_url }}
+    runs-on: ubuntu-latest
+    needs: prepare_page
+    steps:
+      - name: Deploy to GitHub Pages
+        id: deployment
+        uses: actions/deploy-pages@v2
+        with:
+          preview: true
+
```

Comment:
I feel like this looks pretty good, I just don't know these things very well. Looking for general thought process on this, and how to modify it for the purpose of it being hosted on a github page from the core account (github.com/r-selfhosted-wiki).

Reply:
See commit 97c1205d18b7cd84cad52be22bcb0a4971d900be